src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.Tree.cs

```diff
@@ -0,0 +1,449 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class LambdaRewriter
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        internal sealed partial class Analysis : BoundTreeWalkerWithStackGuardWithoutRecursionOnTheLeftOfBinaryOperator
+        {
+            private sealed class Scope
+            {
+                public Scope Parent { get; }
+
+                public ArrayBuilder<Scope> NestedScopes { get; } = ArrayBuilder<Scope>.GetInstance();
+
+                public ArrayBuilder<Closure> Closures { get; } = ArrayBuilder<Closure>.GetInstance();
+
+                public ArrayBuilder<Symbol> DeclaredCapturedVariables { get; } = ArrayBuilder<Symbol>.GetInstance();
```

Comment:
Or DeclaredVariables. It kind of assumed that we are not interested in variables that are not captured.  DeclaredCaptured seems too long

---
In reply to: [126819075](https://github.com/dotnet/roslyn/pull/20715#discussion_r126819075) [](ancestors = 126819075,126740960)